source/content/guides/drupal-9/drupal-9-hosted/07-code.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Migrate a Drupal 8 Site to Drupal 9
+subtitle: Migrate the Code
+description: 
+cms: "Drupal 9"
+categories: [develop]
+tags: [code, launch, migrate, site, updates]
+contributors: [wordsmither]
+reviewed: "2021-03-31"
+layout: guide
+permalink: docs/guides/drupal-9-hosted/code
+anchorid: code
+editpath: hosted/07-code.md
+---
+
+1. From the local Drupal 9 site's directory, use Terminus to retrieve the D8 site's Git URL:
+
+  ```bash{promptUser: user}
+  terminus connection:info $D8_SITE.dev --field=git_url
+  ```
+
+1. Add the Drupal 8 site as a remote repository called `existing-8`. Use the URL retrieved in the previous step:
+
+  ```bash{promptUser: user}
+  git remote add existing-8 ssh://codeserver.dev.xxxx@codeserver.dev.xxxx.drush.in:2222/~/repository.git
+  git fetch existing-8
+  ```
+
+1. Copy over exported configuration from the original site. From your D9 site, run the following commands:
+
+   ```bash{promptUser: user}
+   git checkout existing-8/master -- sites/default/config
+   git mv sites/default/config/* config/
+   git commit -m "Add site configuration."
+   ```
+
+1. Compare your current `pantheon.yml` file with the new D9 `pantheon.upstream.yml`:
+
+  ```bash{promptUser: user}
+  git diff existing-8/master:pantheon.yml pantheon.upstream.yml
+  ```
+
+1. If you have customizations in your D8 site's `pantheon.yml` that you want to keep for D9 (e.g., a Quicksilver script or site-specific protected web paths), copy `pantheon.yml` over:
+
+  ```bash{promptUser: user}
+  git checkout existing-8/master -- pantheon.yml
+  git commit -m "Update pantheon.yml."
+  ```
+
+1. Copy over any Quicksilver scripts referenced in `pantheon.yml`:
+
+  ```bash{promptUser: user}
+  git checkout existing-8/master -- private/scripts
+  git commit -m "Add Quicksilver scripts."
+  ```
+
+1. List contrib modules and themes on your D8 site:
+
+  ```bash{promptUser: user}
+  terminus drush $D8_SITE.dev pm:projectinfo -- --fields=name,version --format=table
+  ```
+
+  The command `pm:projectinfo` assumes Drush 8. If you encounter an issue with this command, [verify and configure the Drush version](/drush-versions) before you continue.
+
+1. Then use Composer on your D9 site to add these there:
+
+  ```bash{promptUser: user}
+  composer require drupal/ctools:^3.4 drupal/redirect:^1.6 drupal/token:^1.7
+  git add composer.*
+  git commit -m "Add contrib projects."
+  ```
+
+1. Copy over any custom modules or themes from your D8 site:
+
+   <Partial file="drupal-9/custom-modules-themes.md" />
```

Comment:
When used to migrate from drops-8, this should NOT have the nested docroot tabs because that's not a common/supported setup